node/impl/full/eth.go

```diff
@@ -570,12 +571,78 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 	return &receipt, nil
 }
 
-func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(context.Context, ethtypes.EthHash, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(ctx context.Context, blkHash ethtypes.EthHash, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	if err != nil {
+		return nil, xerrors.Errorf("block not found")
+	}
+
+	return a.getTransactionByTipsetAndIndex(ctx, ts, index)
 }
 
-func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(context.Context, ethtypes.EthUint64, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(ctx context.Context, blkParam string, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	validBlockTags := map[string]bool{
+		"latest":   true,
+		"pending":  true,
+		"earliest": true,
+		"final":    true,
+	}
+
+	var blockNumber string
+
+	if validBlockTags[blkParam] {
+		blockNumber = blkParam
+	} else {
+		blkNum, err := ethtypes.EthUint64FromHex(blkParam)
+		if err != nil {
+			return nil, fmt.Errorf("invalid block number format: %w", err)
+		}
+		blockNumber = strconv.FormatUint(uint64(blkNum), 10)
+	}
+
+	ts, err := getTipsetByBlockNumber(ctx, a.Chain, blockNumber, true)
+	if err != nil {
+		if err == ErrNullRound {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to get tipset for block %s: %w", blockNumber, err)
+	}
+
+	if ts == nil {
+		return nil, fmt.Errorf("tipset not found for block %s", blockNumber)
+	}
+
+	tx, err := a.getTransactionByTipsetAndIndex(ctx, ts, index)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transaction at index %d: %w", index, err)
+	}
+
+	return tx, nil
+}
+
+func (a *EthAPI) getTransactionByTipsetAndIndex(ctx context.Context, ts *types.TipSet, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	msgs, err := a.Chain.MessagesForTipset(ctx, ts)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get messages for tipset: %w", err)
+	}
+
+	if uint64(index) >= uint64(len(msgs)) {
+		return nil, xerrors.Errorf("index %d out of range: tipset contains %d messages", index, len(msgs))
+	}
+
+	msg := msgs[index]
+
+	cid, err := ts.Key().Cid()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset key cid: %w", err)
+	}
+
+	tx, err := newEthTx(ctx, a.Chain, nil, ts.Height(), cid, msg.Cid(), int(index))
```

Comment:
I think you'll need the `stateTree`, since this API can get any message it's going to encounter `SigTypeSecp256k1` and `SigTypeBLS` messages too. In fact this might be worth checking in an itest because currently it's just looking at eth (delegated) transactions.

Reply:
@virajbhartiya Basically, if you follow the implementation of the ` newEthTx` -> you'll see that it can return "ETH transactions" even for messages whose address is NOT delegated that Filecoin native messages. But to do that -> we need to pass the state tree here. You can see how to get the state tree from other ETH RPC APIs.

Reply:
You can then find an itest somewhere which submits Filecoin transactions and fetch those using these two APIs and ensure it works.